### PR TITLE
feat: add hasura live chat example

### PR DESCRIPTION
**Pull request by alasdairtran**
Hi @jensneuse,

This is an example live chat app using Hasura as a data source for WunderGraph (adapted from [here](https://github.com/hasura/graphql-engine/tree/master/community/sample-apps/realtime-chat)). I'm marking this as draft as I mainly want to show a couple of corner cases that WunderGraph currently fails to handle.

### 1. Some queries with complex `where` clauses fail validation

Two examples are the `FetchNewMessages` query [here](https://github.com/mochilabs/wundergraph/blob/hasura-example/examples/graphql-hasura-live-chat/app/.wundergraph/operations/FetchNewMessages.graphql) and the `SubscribeToUserTyping` subscription [here](https://github.com/mochilabs/wundergraph/blob/hasura-example/examples/graphql-hasura-live-chat/app/.wundergraph/operations/SubscribeToUserTyping.graphql). They both give a fairly opaque `mismatched input value` error during validation when starting WunderGraph. In those two graphql files, I also provide in the comments simpler queries that pass validation.

### 2. Queries containing Boolean input pass validation but fails to return data

The `FetchAllMessages` query [here](https://github.com/mochilabs/wundergraph/blob/hasura-example/examples/graphql-hasura-live-chat/app/.wundergraph/operations/FetchAllMessages.graphql) passes validation (I believe thanks to https://github.com/wundergraph/graphql-go-tools/pull/447) *but* fails to return any data. I console.log the response [here](https://github.com/mochilabs/wundergraph/blob/hasura-example/examples/graphql-hasura-live-chat/app/components/chat/RenderMessages.tsx#L53) and it's always null.

Let me know if this PR is enough for you guys to reproduce, and if you'd like me to create separate issues to track them.

### CODE REVIEW

**Comment by jensneuse** [comment]
Thank you @alasdairtran! We will have a look. @spetrunin can you try this out with the latest engine?

**Comment by alasdairtran** [comment]
Hi @spetrunin, just wondering if you had the chance to reproduce these errors?

**Comment by spetrunin** [comment]
Hi @alasdairtran
yep, we have reproduced it
currently working on a fix

File: examples/graphql-hasura-live-chat/app/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "scripts": {
+    "start": "run-p dev wundergraph",
+    "wundergraph": "wunderctl up --debug",
+    "open": "wait-on -d 500 http://localhost:9991 && open-cli http://localhost:3000",
+    "build": "next build",
+    "dev": "next dev --port 3000",
+    "check": "tsc --noEmit"
+  },
+  "dependencies": {
+    "@wundergraph/sdk": "^0.123.2",
```
Comment (spetrunin) [inline examples/graphql-hasura-live-chat/app/package.json:14]: next sdk release will include a fix to make this example work

File: examples/graphql-hasura-live-chat/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+version: '3.9'
+
+services:
+  postgres:
+    image: docker.io/bitnami/postgresql:14.4.0
```
Comment (spetrunin) [inline examples/graphql-hasura-live-chat/docker-compose.yml:5]: could we use official postgres image instead to been able to launch examples on m1 mac?

File: examples/graphql-hasura-live-chat/hasura/console.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+FROM debian:buster-slim
+
+RUN apt-get update -y \
+ && apt-get install -y curl socat \
+ && curl -L -o /usr/local/bin/hasura https://github.com/hasura/graphql-engine/releases/download/v2.15.0/cli-hasura-linux-amd64 \
```
Comment (spetrunin) [inline examples/graphql-hasura-live-chat/hasura/console.Dockerfile:5]: could you add commented line with arm version also and add a note to readme about that?

File: examples/graphql-hasura-live-chat/hasura/engine.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Build instructions:
+#   docker build -t 159213555475.dkr.ecr.ap-southeast-2.amazonaws.com/hasura-engine:v2.15.0 -f storage/hasura/engine.Dockerfile storage/hasura
+#   docker push 159213555475.dkr.ecr.ap-southeast-2.amazonaws.com/hasura-engine:v2.15.0
+FROM hasura/graphql-engine:v2.15.0
```
Comment (spetrunin) [inline examples/graphql-hasura-live-chat/hasura/engine.Dockerfile:4]: hasura release `v2.15.0` was removed, nearest available version is `v2.15.2`

**Review by spetrunin — CHANGES_REQUESTED** [review]

File: examples/graphql-hasura-live-chat/app/package.json (hunk 1 above)
Comment (spetrunin) [reply examples/graphql-hasura-live-chat/app/package.json:14]: @alasdairtran I will notify you once it is released

**Comment by spetrunin** [comment]
@alasdairtran this release `@wundergraph/sdk@0.129.0` contains required fixes

we still have to do some work to properly integrate this example into our CI pipeline

please let us know in case you have any other problems
thanks again for your input!

**Pull request closed without merging**